src/common/lib/python/common_task.h

```diff
@@ -0,0 +1,137 @@
+    PyObject *arguments,
+    std::unordered_map<std::string, double>& required_resources,
+    std::function<PyObject *(PyObject *)> callMethodObjArgs,
+    std::function<bool(PyObject *)> isPyObjectIDType);
+  NonRayletTask(const char* data, int data_size);
+  NonRayletTask(TaskSpec *task_spec, int64_t task_size);
+  ~NonRayletTask();
+  PyObject *to_string();
+  FunctionID function_id();
+  ActorID actor_id();
+  int64_t actor_counter();
+  UniqueID driver_id();
+  TaskID task_id();
+  TaskID parent_task_id();
+  int64_t parent_counter();
+  int64_t num_args();
+  int arg_id_count(int index);
+  ObjectID arg_id(int64_t arg_index, int64_t id_index);
+  const uint8_t *arg_val(int64_t arg_index);
+  int64_t arg_length(int64_t arg_index);
+  ActorID actor_creation_id();
+  ObjectID actor_creation_dummy_object_id();
+  std::unordered_map<std::string, double> get_required_resources();
+  int64_t num_returns();
+  ObjectID return_id(int64_t return_index);
+  void to_serialized_flatbuf(flatbuffers::FlatBufferBuilder &fbb,
+    std::vector<ray::ObjectID> execution_dependencies);
+  void to_submit_message(flatbuffers::FlatBufferBuilder &fbb,
+    const std::vector<ObjectID> &execution_dependencies);
+private:
+  int64_t size;
+  // The task spec to use in the non-raylet case.
+  TaskSpec *spec;
+};
+
+class RayletTask : public TaskInterface {
+public:
+  RayletTask(UniqueID &driver_id, TaskID &parent_task_id, int parent_counter,
+    ActorID &actor_creation_id, ObjectID &actor_creation_dummy_object_id, 
+    UniqueID &actor_id, UniqueID &actor_handle_id, int actor_counter, 
+    bool is_actor_checkpoint_method, FunctionID &function_id, int num_returns,
+    PyObject *arguments,
+    std::unordered_map<std::string, double>& required_resources,
+    std::function<PyObject *(PyObject *)> callMethodObjArgs,
+    std::function<bool(PyObject *)> isPyObjectIDType);
+  PyObject *to_string();
+  FunctionID function_id();
+  ActorID actor_id();
+  int64_t actor_counter();
+  UniqueID driver_id();
+  TaskID task_id();
+  TaskID parent_task_id();
+  int64_t parent_counter();
+  int64_t num_args();
+  int arg_id_count(int index);
+  ObjectID arg_id(int64_t arg_index, int64_t id_index);
+  const uint8_t *arg_val(int64_t arg_index);
+  int64_t arg_length(int64_t arg_index);
+  ActorID actor_creation_id();
+  ObjectID actor_creation_dummy_object_id();
+  std::unordered_map<std::string, double> get_required_resources();
+  int64_t num_returns();
+  ObjectID return_id(int64_t return_index);
+  void to_serialized_flatbuf(flatbuffers::FlatBufferBuilder &fbb,
+    std::vector<ray::ObjectID> execution_dependencies);
+  void to_submit_message(flatbuffers::FlatBufferBuilder &fbb,
+    const std::vector<ObjectID> &execution_dependencies);
+private:
+  // The task spec to use in the raylet case.
+  std::unique_ptr<ray::raylet::TaskSpecification> task_spec;
+};
+
+// clang-format off
+typedef struct {
```

Comment:
why struct, instead of class as well?

Reply:
I copied from the original file, not sure class work for Python interface or not